### ISSUE TITLE
Consider supporting touchscreens a bit better

### ISSUE THREAD

**Issue by turt2live** [vector-im/riot-web#13526]
One case where this goes wrong is that context menus open where the mouse is instead of where the long press was

**Comment by WillPittenger** [comment]
I also just noticed another case: I saw the red underlining for a misspelled word.  Since I'm now running version 1.6 of the Windows desktop client, I tried to long tap to get a list of suggestions.  But Riot is treating my long tap as a double click and selecting the word.  The menu listing suggestions never appears.

**Comment by WillPittenger** [comment]
Not sure if this should be a meta issue with separate child issues for each of touch screen bug.

**Comment by ell1e** [comment]
Fwiw, on the PinePhone this is the only way  to use Element right now and it's kind of rough but almost there/very promising. Here is what I ran into:

- there seems to be no way to shrink the rooms/chats sidebar without a mouse, even when in portrait orientation it only leaves a ridiculously tiny column for actual conversation text just to show room names lengthily
- popup menu with reply, react, new threads buttons should show after long press on message, not some firefox popup menu
- popup menu buttons (reply/react/new threads) randomly fail to work via touch, resulting in no effect after pressing other than the menu closing again. they should also imho have more horizontal spacing to make it easier to hit the right one

**Comment by Half-and-Xat** [comment]
Coming from Librem 5 and installed the Debian version of Element. Everything works fine except what @ell1e said about shrinking rooms/chats and also except scrolling via scroll bar. 

A short touch of touchscreen is a left click. A long touch is a right click (hold down). But scrolling bar and resizing rooms/chats can only be done with long left click. This way touch input has no access to that slider elements. 

However, it is also inconsistent between different slider elements. On the rooms panel there is a slider below the rooms that can hide rooms above, if you slide it up. You can do that with left **and** with right click (and drag up). Implementing same to scroll bar and rooms/chats horizontal slider would make the overall design more consistent, but also would fix touch input issues.

The scroll bar is a low priority issue since there is also another way to scroll (drag the chat panel up and down), but to shrink the rooms/chats is more critical. People who can't work around this have not enough horizontal space on small screens (720px horizontal or even less) to read the chat without troubles.

**Comment by Half-and-Xat** [comment]
I want to add additional and up-to-date information about the slider elements. 

Functional sliders:
- Slider to shrink and expand room count (horizontal slider below rooms)
- Slider between chat and opened right panel (new info)

Dysfunctional sliders:
- Slider between chat and room panel
- Slider between calls panel and chat (new info)

Sliders that are somehow functional in a buggy way
- All scrolling bars (updated info)

The Slider between chat and room panel is still the most important one that need a fix. It prevents people from using Element on pocket computers (until they get support on our forums to shrink it in a hacky way on each start). 

Scrolling bars can be touched and dragged. However, while using a mouse I can click on the bar and drag outside, the scrolling bar will still scroll to the right position - on touch I will lose the focus out of bar (scrolling stops). That makes it hard to use. On chat scrolling bar it even jumps between the position I'm touching and somewhere else. It's that hard to use, that it is kinda pointless to even try on daily use.

**Comment by t3chguy** [comment]
> All scrolling bars (updated info)

The scroll bars are just native browser elements, we don't control their behaviour only their width

**Comment by Half-and-Xat** [comment]
Indeed, I tried it on Firefox and there it works fine. So is this an issue on electron app itself and should I create an issue there? Or is this an issue that has to be fixed on system side?

**Comment by t3chguy** [comment]
Electron or more likely Chromium
